tools/deployment/presto-clp/scripts/init.py

```
@@ -363,6 +377,122 @@ def _add_worker_env_vars(coordinator_common_env_file_path: Path, env_vars: dict[
+
+
+def _image_exists_locally(docker_executable: str, image_ref: str) -> bool:
+    """
+    Returns whether `image_ref` is present in the local Docker daemon.
+
+    :param docker_executable: The Docker executable to call.
+    :param image_ref: The image reference (repository:tag).
+    :return: True if the image exists locally.
+    """
+    return (
+        subprocess.run(
```

Ruff (0.16.0)

[error] `subprocess` call: check for execution of untrusted input

(S603)